### PR TITLE
Better docker cntd

### PR DESCRIPTION
**Pull request by kba**
Cherry-pick of @zaventh's improvements to docker setup against current master. See #110

### CODE REVIEW

**Comment by juanpicado** [comment]
Thanks @kba I'll review this soon.

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-RUN mkdir -p $appdir
-WORKDIR  $appdir
-COPY . $appdir
+RUN mkdir -p "$APPDIR"
+WORKDIR $APPDIR
+COPY . $APPDIR
 RUN npm install
 
 RUN mkdir -p /verdaccio/storage /verdaccio/conf
 ADD conf/docker.yaml /verdaccio/conf/config.yaml
```
Comment (juanpicado) [inline Dockerfile]: According this, COPY is better than ADD . Are you agreed ? https://www.codacy.com/app/juanpicado/verdaccio/issues?&filters=W3siaWQiOiJMYW5ndWFnZSIsInZhbHVlcyI6WyJEb2NrZXJmaWxlIl19LHsiaWQiOiJDYXRlZ29yeSIsInZhbHVlcyI6W251bGxdfSx7ImlkIjoiTGV2ZWwiLCJ2YWx1ZXMiOltudWxsXX0seyJpZCI6IlBhdHRlcm4iLCJ2YWx1ZXMiOltudWxsXX0seyJpZCI6IkF1dGhvciIsInZhbHVlcyI6W251bGxdfV0=

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-ENV appdir /usr/local/app
+ENV APPDIR /usr/local/app
 
-RUN mkdir -p $appdir
-WORKDIR  $appdir
-COPY . $appdir
+RUN mkdir -p "$APPDIR"
+WORKDIR $APPDIR
+COPY . $APPDIR
 RUN npm install
```
Comment (juanpicado) [inline Dockerfile]: Could you add the MAINTAINER ? I guess the name of the project would be enough, unless :) you want to add your own email

Comment (kba) [inline Dockerfile]: I think the `MAINTAINER` field has been [deprecated](https://docs.docker.com/engine/reference/builder/#maintainer-deprecated). Maybe

```dockerfile
LABEL maintainer="https://github.com/verdaccio/verdaccio"
```

File: Dockerfile (hunk 1 above)
Comment (kba) [inline Dockerfile]: Right, `ADD` supports copying from a URL or archive, though I wonder who ever uses that. I'll change it.

File: Dockerfile (hunk 2 above)
Comment (juanpicado) [inline Dockerfile]: It seems good for me

**Comment by juanpicado** [comment]
@kba I'm having the same issue that I posted at #110 . 

````bash
$> docker build -t verdaccio-build . 
Sending build context to Docker daemon 122.7 MB
Step 1/14 : FROM node:6-alpine
6-alpine: Pulling from library/node
79650cf9cc01: Pull complete 
db515f170158: Pull complete 
559bb4ad61e1: Pull complete 
Digest: sha256:50926ed1350d4ba539ba5bf72e88f6113436cb8394b195f36fc8397fc5bc3a70
Status: Downloaded newer image for node:6-alpine
 ---> 1017e766f05b
Step 2/14 : LABEL maintainer "https://github.com/verdaccio/verdaccio"
 ---> Running in 2dfe832d82f3
 ---> d0364630427c
Removing intermediate container 2dfe832d82f3
Step 3/14 : ENV APPDIR /usr/local/app
 ---> Running in 28b1b6ba8686
 ---> f4ce1e6f60f1
Removing intermediate container 28b1b6ba8686
Step 4/14 : RUN mkdir -p "$APPDIR"
 ---> Running in 044fd5b54f9e
 ---> ad437bab829c
Removing intermediate container 044fd5b54f9e
Step 5/14 : WORKDIR $APPDIR
 ---> c29e1a04d76c
Removing intermediate container effaf11b3aa4
Step 6/14 : ADD . $APPDIR
 ---> b6105fa7d2e6
Removing intermediate container 707242d2ca76
Step 7/14 : RUN npm install
 ---> Running in 766aa72c02ef
npm info it worked if it ends with ok
npm info using npm@3.10.10
npm info using node@v6.10.3
npm info lifecycle verdaccio@2.1.7~preinstall: verdaccio@2.1.7
npm info linkStuff verdaccio@2.1.7
npm info lifecycle verdaccio@2.1.7~install: verdaccio@2.1.7
npm info lifecycle verdaccio@2.1.7~postinstall: verdaccio@2.1.7
npm info lifecycle verdaccio@2.1.7~prepublish: verdaccio@2.1.7
npm info ok 
 ---> 53a3ed7e36b1
Removing intermediate container 766aa72c02ef
Step 8/14 : RUN mkdir -p /verdaccio/storage /verdaccio/conf
 ---> Running in 750c7f6b19b2
 ---> d16b95f82c7d
Removing intermediate container 750c7f6b19b2
Step 9/14 : ADD conf/docker.yaml /verdaccio/conf/config.yaml
 ---> 05a138f47b7b
Removing intermediate container 7e23817ad30e
Step 10/14 : RUN addgroup -S verdaccio && adduser -S -g verdaccio verdaccio &&     chown -R verdaccio:verdaccio "$APPDIR" &&     chown -R verdaccio:verdaccio /verdaccio
 ---> Running in e9e67c98fbd5
````
Shouldn't it works ? I'm missing any command?

**Comment by juanpicado** [comment]
I ran with `docker-compose up --build` and I got the same result.

**Comment by juanpicado** [comment]
Well, actually it does not get hang, it takes 10 minutes to apply 

```
RUN chown -R --changes verdaccio:verdaccio "$APPDIR"
```
to the whole `node_modules` folder.

**Comment by juanpicado** [comment]
Approving this, the improvements are quite noticeable.

* Decrease the image size from `754 MB` to `228 MB`. 

```bash
verdaccio/local                latest              9859165f29e9        12 hours ago        228 MB
verdaccio/verdaccio            latest              7d6b55a2279f        9 days ago          754 MB
````
* Run verdaccio process with his own user
```
PID   USER     TIME   COMMAND
    1 verdacci   0:00 sh -c ${APPDIR}/bin/verdaccio --config /verdaccio/conf/co
    6 verdacci   0:00 verdaccio
   12 verdacci   0:00 sh
   17 verdacci   0:00 ps -ef
```
* Use `node-alpine`
* As optional `docker-compose` definition file

**Pull request merged**